### DISCUSSION TITLE
nc-automount and other scripts missing during NextCloudPi setup on Proxmox LXC

### DISCUSSION THREAD

**Discussion by WaldemarFech** [community-scripts/ProxmoxVE#163] [Q&A] [answered]
Hello,

First of all, thank you for providing this helpful script and for all your work on this project.

I'm encountering an issue where several scripts, such as **nc-automount** and **nc-format-USB**, are missing during the setup of NextCloudPi on Proxmox using the LXC installation script from [community-scripts](https://community-scripts.github.io/ProxmoxVE/scripts?id=nextcloudpi). Additionally, I receive an **"nc-automount read error"** during the setup wizard.

**Steps I've followed:**

1. Ran the installation command:
   ```bash
   bash -c "$(wget -qLO - https://github.com/community-scripts/ProxmoxVE/raw/main/ct/nextcloudpi.sh)"
   ```
2. During installation, I went to **Advanced** and manually entered the IP address, vmbr, and DNS IP. The rest of the settings were left as default.
3. Tried both **privileged** and **unprivileged** containers, but the issue persists in both cases.
4. Once NextCloudPi is running, I backed up the two NCP passwords and clicked on **Activate**.
5. After activation, I'm redirected to the admin login page. I log in using **ncp** and the corresponding password.
6. The setup wizard appears, and I start it.
7. When asked if I use USB, I select **Yes** and click **Continue**.
8. Immediately, I receive an orange error message at the bottom right: **"nc-automount read error"**.

**Additional observations:**

- On the admin web UI (`:4443`), under the **Tools** section, many entries seem to be missing—specifically **nc-automount** and **nc-format-USB**—which are mentioned in the wiki.
- I shut down the LXC container, then in Proxmox under **Resources**, I added a device passthrough and selected `/dev/sdc` with access mode in CT set to `0660`. Unfortunately, this didn't resolve the issue.

**Output of `lsblk` inside the container:**

```bash
root@nextcloudpi:~# lsblk
NAME     MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS
sda        8:0    0  59.6G  0 disk 
├─sda1     8:1    0  1007K  0 part 
├─sda2     8:2    0   512M  0 part 
└─sda3     8:3    0  59.1G  0 part 
sdb        8:16   0 476.9G  0 disk 
├─sdb1     8:17   0  1007K  0 part 
├─sdb2     8:18   0     1G  0 part 
└─sdb3     8:19   0  58.6G  0 part 
sdc        8:32   0   4.5T  0 disk 
zd0      230:0    0   4.5G  0 disk 
zd16     230:16   0    10G  0 disk 
├─zd16p1 230:17   0   512K  0 part 
├─zd16p2 230:18   0     1G  0 part 
└─zd16p3 230:19   0     9G  0 part
```

I've been trying different variations all day without success, and I'm not sure what I'm missing.

Could you please help me understand what might be causing this issue and how to resolve it?

Thank you in advance for your assistance.

Best regards,
Waldemar

![image](https://github.com/user-attachments/assets/3843b067-ed1a-4e35-b541-168bba5ddb23)

**Comment by Mellowlynx** [comment]
If I understand correctly, this option is not available by design.
As a LXC does not have access to hardware resources, It can't look for USB and auto-mount one.
Proxmox should handle the mount to the LXC, and you can set up the mount location within the UI. (_nc-datadir_ probably)
I did it this way for a separate SSH, and it should also work with a USB drive.

**Reply by WaldemarFech** [reply]
Thanks for the info!

But even if auto-mounting isn't possible in an LXC, shouldn't the `nc-format-USB` option still show up in the NextCloudPi admin panel? The USB hard drive is actually recognized in the system info window, so the system seems to see it.

Do you think it might be better to set up NextCloudPi in a VM instead of an LXC container to get all features working, like USB auto-mounting and formatting? Or does a VM have even more limitations? And shouldn't the LXC with device passthrough still work?

Really appreciate your help!

![image](https://github.com/user-attachments/assets/b9bccae2-6a28-49c9-83e6-4b2f9bdf37c6)

**Reply by Mellowlynx** [reply, marked as answer]
I can't give a final answer if it should be there, I did read that it's not included in the docker image. But I'm sure that the option is just removed, not all messages, that would be to match of a hassle.

I also run it in a LXC, and it runs great, I added an extra SSD to the host and added a mount point to a .raw disk on that SSD.
For what you want, it might be better to have it as a VM, I agree with that.

Passthrough to a LXC is not something I use, so match problems.
VM's are better with passthrough, I do use in on my Home Assistant VM, and there it works great.

So maybe for you, it's worth the bot more overhead of running a VM, but having an easier setup.
